### ISSUE TITLE
Report error if `|` union operator is used with a string

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3244]
Python 3.10 doesn't allow part of a type annotation union to be a forward declaration, as in:

```python
class Foo:
    x: None | "Foo" = None # TypeError: unsupported operand type(s) for |: 'NoneType' and 'str'
```

Pyright should detect and report this error.

**Comment by erictraut** [comment]
This turns out to be much more difficult than I expected. The Python runtime is very inconsistent about which types support strings and which do not. For example, `A | "B"` is an error but `A[int] | "B"` and `Optional[A] | "B"` is not an error. So implementing this would require that we reverse engineer the runtime — and track its changes from one version to the next. I don't think it's worth it to detect this problem, which is a bit of an edge case.

**Comment by debonte** [comment]
In case others run into this, you can work around it by wrapping the whole union in quotes:
```python
class Foo:
    x: "None | Foo" = None
```
